src/async-tokenize/tests/async-tokenize.test.ts

```diff
@@ -0,0 +1,289 @@
+import { fireEvent, waitFor } from '@testing-library/dom'
+import {
+  formElementsMock,
+  formValuesMock,
+  handleFormMock,
+  malgaConfigurations,
+  configureFormSubmissionMock,
+} from '../../../tests/mocks/malga-tests-mocks'
+import { Malga } from '../../common/malga/malga'
+import { AsyncTokenize } from '../async-tokenize'
+import * as utilsValues from '../../common/utils/form-values/form-values'
+import * as utilsElements from '../../common/utils/form-elements/form-elements'
+
+const onSubmit = vi.fn()
+
+vi.mock('../../common/malga/malga', async (importOriginal) => {
+  const Malga =
+    await importOriginal<typeof import('../../common/malga/malga')>()
+  return {
+    ...Malga,
+    tokenization: vi.fn(),
+  }
+})
+
+function generateForm(onSubmit: any) {
+  const { form, holderNameInput, cvvInput, expirationDateInput, numberInput } =
+    handleFormMock()
+
+  form.setAttribute(formElementsMock.form, '')
+  form.onsubmit = onSubmit
+  form.id = 'form'
+  form.method = 'POST'
+  form.action = '/test'
+
+  holderNameInput.setAttribute(formElementsMock.holderName, '')
+  numberInput.setAttribute(formElementsMock.number, '')
+  cvvInput.setAttribute(formElementsMock.cvv, '')
+  expirationDateInput.setAttribute(formElementsMock.expirationDate, '')
+
+  document.body.appendChild(form)
+  form.appendChild(holderNameInput)
+  form.appendChild(numberInput)
+  form.appendChild(expirationDateInput)
+  form.appendChild(cvvInput)
+
+  const inputs = document.querySelectorAll('input')
+  inputs[0].value = formValuesMock.holderName
+  inputs[1].value = formValuesMock.number
+  inputs[2].value = formValuesMock.expirationDate
+  inputs[3].value = formValuesMock.cvv
+}
+
+describe('handle', () => {
```

Comment:
Aqui seria melhor ter uma mensagem mais descritiva, geralmente o primeiro `describe` é o nome do arquivo.

```suggestion
describe('async-tokenize', () => {
```